src/js/components/Tree/Tree.js

```diff
@@ -0,0 +1,161 @@
+import React, { useContext, useState, useEffect } from 'react';
+import { ThemeContext } from 'styled-components';
+import { FormDown } from 'grommet-icons/icons/FormDown';
+import { FormNext } from 'grommet-icons/icons/FormNext';
+import { FormPrevious } from 'grommet-icons/icons/FormPrevious';
+
+import { Box } from '../Box';
+import { Button } from '../Button';
+import { List } from '../List';
+import { Text } from '../Text';
+
+const Tree = ({ children, mode, data }) => {
+  const theme = useContext(ThemeContext);
+
+  const [depth, setDepth] = useState(0);
+  // an array of the path names
+  const [breadcrumbs, setBreadcrumbs] = useState([]);
+  const [selectedNode, setSelectedNode] = useState();
+  // treeLevel should only be updated on useEffect to reflect correct depth
+  const [treeLevel, setTreeLevel] = useState(data);
+  // parentLevel should only be updated on useEffect to reflect correct depth
+  const [parentLevel, setParentLevel] = useState();
+
+  useEffect(() => {
+    let level = data;
+    let parent;
+    for (let i = 0; i < depth; i += 1) {
+      // follow the breadcrumbs path
+      parent = level.find(element => element.name === breadcrumbs[i]);
+      level = parent.children;
+    }
+    setParentLevel(parent);
+    setTreeLevel(level);
+  }, [breadcrumbs, data, depth]);
+
+  const ColumnNavigationList = () => (
+    <Box width="medium">
+      <List
+        data={treeLevel}
+        onClickItem={event => setSelectedNode(event.item)}
+        {...theme.tree.column.listProps}
+      >
+        {datum => {
+          return (
+            <Box direction="row" justify="between" align="center" gap="small">
+              <Text size="small">{datum.name}</Text>
+              <Box direction="row" gap="hair" align="center">
+                <Text size="xsmall">
```

Comment:
This part, where we see the count and the label "sub-resources" feels too opinionated to me. I'm inclined to not include this in our first go at Tree.